source/server/server_configuration_parser.cpp

```diff
@@ -237,11 +221,62 @@ std::string ServerConfigurationParser::read_keycert(const std::string& filename)
   return data;
 }
 
+std::string ServerConfigurationParser::parse_bind_address() const
+{
+  // Use default address if none is specified
+  std::string parsed_bind_address = kDefaultAddressPrefix;
+
+  auto it = config_file_.find(kAddressJsonKey);
+  if (it != config_file_.end()) {
+    try {
+      parsed_bind_address = it->get<std::string>();
+    }
+    catch (const nlohmann::json::type_error& ex) {
+      throw WrongPortTypeException(ex.what());
```

Comment:
InvalidAddressException ?

Reply:
Throw WrongAddressTypeException if the bind address is not a string. Added unit tests for use case.